src/System.Xml.XDocument/tests/SDMSample/SDMDocument.cs

```diff
@@ -4,189 +4,146 @@
-                    Validate.IsNull(doc.Root);
-                    Validate.IsNull(doc.Declaration);
-                    Validate.Enumerator(doc.Nodes(), new XNode[0]);
-                }
-
-                /// <summary>
-                /// Validate behavior of the XDocument constructor that takes content.
-                /// </summary>
-                /// <returns>true if pass, false if fail</returns>
-                //[Variation(Desc = "CreateDocumentWithContent")]
-                public void CreateDocumentWithContent()
-                {
-                    XDeclaration declaration = new XDeclaration("1.0", "utf-8", "yes");
-                    XComment comment = new XComment("This is a document");
-                    XProcessingInstruction instruction = new XProcessingInstruction("doc-target", "doc-data");
-                    XElement element = new XElement("RootElement");
-
-                    XDocument doc = new XDocument(declaration, comment, instruction, element);
-
-                    Validate.Enumerator(
-                        doc.Nodes(),
-                        new XNode[] { comment, instruction, element });
-                }
-
-                /// <summary>
-                /// Validate behavior of the XDocument copy/clone constructor.
-                /// </summary>
-                /// <returns>true if pass, false if fail</returns>
-                //[Variation(Desc = "CreateDocumentCopy")]
-                public void CreateDocumentCopy()
-                {
-                    try
-                    {
-                        new XDocument((XDocument)null);
-                        Validate.ExpectedThrow(typeof(ArgumentNullException));
-                    }
-                    catch (Exception ex)
-                    {
-                        Validate.Catch(ex, typeof(ArgumentNullException));
-                    }
-
-                    XDeclaration declaration = new XDeclaration("1.0", "utf-8", "yes");
-                    XComment comment = new XComment("This is a document");
-                    XProcessingInstruction instruction = new XProcessingInstruction("doc-target", "doc-data");
-                    XElement element = new XElement("RootElement");
-
-                    XDocument doc = new XDocument(declaration, comment, instruction, element);
-
-                    XDocument doc2 = new XDocument(doc);
-
-                    IEnumerator e = doc2.Nodes().GetEnumerator();
-
-                    // First node: declaration
-                    Validate.IsEqual(doc.Declaration.ToString(), doc2.Declaration.ToString());
-
-                    // Next node: comment
-                    Validate.IsEqual(e.MoveNext(), true);
-                    Validate.Type(e.Current, typeof(XComment));
-                    Validate.IsNotReferenceEqual(e.Current, comment);
-                    XComment comment2 = (XComment)e.Current;
-                    Validate.IsEqual(comment2.Value, comment.Value);
```

Comment:
`Validate.IsEqual` verified that neither of the both parameters is `null`, however `Assert.Equal` with which it was replaced doesn't.

You might want to audit each of those and make sure to add `Assert.NotNull` where appropriate (e.g. most of the time one of the two arguments is known to be non-null so it's not necessary there).